shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+  ///             CommitUpdates().
+  ///
+  /// @param[in]  action           A pointer to the custom semantics action
+  /// update.
+  void AddFlutterSemanticsCustomActionUpdate(
+      const FlutterSemanticsCustomAction* action);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Flushes the pending updates and applies them to this
+  ///             accessibility bridge.
+  void CommitUpdates();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the underlying AXTree.
+  AXTree* GetAXTree();
+
+  //------------------------------------------------------------------------------
+  /// @brief      The event generator of this accessibility bridge. It contains
+  ///             the pending accessibility events generated as a result of a
+  ///             semantics update.
+  AXEventGenerator* GetEventGenerator();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the user data.
+  void* GetUserData();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the accessibility bridge delegate.
+  AccessibilityBridgeDelegate* GetDelegate();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the flutter accessibility node with the given id from this
+  ///             accessibility bridge.
+  ///
+  /// @param[in]  id           The id of the flutter accessibility node you want
+  ///                          to retrieve.
+  FlutterAccessibility* GetFlutterAccessibilityFromID(int32_t id) const;
+
+  //------------------------------------------------------------------------------
+  /// @brief      Update the currently focused flutter accessibility node.
+  ///
+  /// @param[in]  id           The id of the currently focused flutter
+  ///                          accessibility node.
+  void SetFocusedNode(int32_t node_id);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the last focused node.
+  int32_t GetLastFocusedNode();
+
+  // AXTreeObserver implementation.
+  void OnNodeWillBeDeleted(AXTree* tree, AXNode* node) override;
+  void OnSubtreeWillBeDeleted(AXTree* tree, AXNode* node) override;
+  void OnNodeCreated(AXTree* tree, AXNode* node) override;
+  void OnNodeDeleted(AXTree* tree, int32_t node_id) override;
+  void OnNodeReparented(AXTree* tree, AXNode* node) override;
+  void OnRoleChanged(AXTree* tree,
+                     AXNode* node,
+                     ax::mojom::Role old_role,
+                     ax::mojom::Role new_role) override;
+  void OnAtomicUpdateFinished(
+      AXTree* tree,
+      bool root_changed,
+      const std::vector<AXTreeObserver::Change>& changes) override;
+
+  std::unique_ptr<AccessibilityBridgeDelegate> delegate_;
+
+ private:
+  // See FlutterSemanticsNode in embedder.h
```

Comment:
Why are we redefining these here? Why not just use them from embedder.h?

If we do this, we have these definitions in embedder.h, semantics_node.h, and here.

Reply:
There are two reason i did this.

1. the embedding sends update to a semantics node one by one, and the memory of the semantics node is destroy right before it sends the next one. That means I have to create a memory copy of the the FlutterSemanticsNode, which means i have to write a function to assign each attribute one by one anyway.
2. continuing on 1, the FlutterSemanticsNode uses raw pointer to hold data array, which is make it a bit risky to manage the memory allocation. If I create a different struct to use std vector, it will be easier to manage the memory allocation.